File: app/src/main/java/net/ducksmanager/persistence/dao/UserSettingDao.java
package net.ducksmanager.persistence.dao;

import net.ducksmanager.persistence.models.composite.UserSetting;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface UserSettingDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(UserSetting userSetting);

    @Query("SELECT user_settings.* FROM user_settings WHERE settingKey = :key")
    UserSetting findByKey(String key);

    @Query("DELETE FROM user_settings")
    void deleteAll();
}

File: app/src/main/java/net/ducksmanager/persistence/dao/UserSettingDao.kt
package net.ducksmanager.persistence.dao

import androidx.annotation.Nullable
import androidx.room.*
import net.ducksmanager.persistence.models.composite.UserSetting

@Dao
interface UserSettingDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(userSetting: UserSetting)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun update(@Nullable userSetting: UserSetting?)

    @Query("SELECT user_settings.* FROM user_settings WHERE settingKey = :key")
    fun findByKey(key: String): UserSetting?

    @Query("DELETE FROM user_settings")
    fun deleteAll()
}